src/shared/extension-release.c

```diff
@@ -119,14 +119,131 @@ int extension_release_validate(
         return 1;
 }
 
-int parse_env_extension_hierarchies(char ***ret_hierarchies) {
+int configuration_release_validate(
+                const char *name,
+                const char *host_os_release_id,
+                const char *host_os_release_version_id,
+                const char *host_os_release_syscfg_level,
+                const char *host_syscfg_scope,
+                char **configuration_release) {
```

Comment:
instead of copy/pasting this function, please rework the existing function to be `extension_release_validate_internal()` or so, that gets strings like "SYSCFG_SCOPE"/"SYSEXT_SCOPE" and "SYSCFG_LEVEL"/"SYSEXT_LEVEL=" as additional parameters.

Then provide two wrapper calls around extension_release_validate_internal() that pass all parameters as is, and set these extra parameters.